doc/tutorial/single_cell_recipe.rst

```diff
@@ -0,0 +1,181 @@
+
+        def num_sources(self, gid):
+            # (4.3) Override the num_sources method
+            return 1
+
+        def cell_kind(self, gid):
+            # (4.4) Override the cell_kind method
+            return arbor.cell_kind.cable
+
+        def cell_description(self, gid):
+            # (4.5) Override the cell_description method
+            return self.the_cell
+
+        def probes(self, gid):
+            # (4.6) Override the probes method
+            return self.the_probes
+
+        def global_properties(self, kind):
+            # (4.7) Override the global_properties method
+            return self.the_props
+
+Step **(4)** describes the recipe that will reflect our single cell model.
+
+Step **(4.1)** defines the class constructor. It can take any shape you need, but it is important to call base class' constructor. In general it's a good idea to make objects that'll be return by methods that need to be overridden members of the class. With this constructor, we could easily change the cell and probes of the model, should be want to do so. Here we initialize the cell properties to match Neuron's defaults using Arbor's built-in :meth:`arbor.neuron_cable_properties` and extend with Arbor's own :meth:`arbor.default_catalogue`.
+
+Step **(4.2)** defines that this model has one cell.
+
+Step **(4.3)** defines that this model has one source.
+
+Step **(4.4)** returns the :class:`arbor.cell_kind.cable`. Make sure you don't instantiate the model with a non cable cell! Since we have just one cell, we can ignore :gen:`gid`.
+
+Step **(4.5)** returns the cell description passed in on class initialisation. If you would have multiple cells of different kinds, make sure the right types are returned when :meth:`arbor.recipe.cell_kind` and :meth:`arbor.recipe.cell_description` are called with the same :gen:`gid`.
+
+Step **(4.6)** returns the probes passed in at class initialisation.
+
+Step **(4.7)** returns the properties that will be applied to all cells of that kind in the model.
+
+More methods can be overridden if your model requires that, see :class:`arbor.recipe` for options.
+
+Step **(5)** instantiates the recipe with the cable cell described earlier, and a single voltage probe located at "center".
+
+The context and domain decomposition
+------------------------------------
+
+:class:`arbor.single_cell_model` does not only take care of the recipe, it also takes care of defining how the simulation will be ran. When you create and use your own recipe, you'll need to do this manually, in the form of defining a hardware context and a domain decomposition. Fortunately, the default constructors of :class:`arbor.context` and :class:`arbor.partition_load_balance` are sufficient for this model, and is what :class:`arbor.single_cell_model` does under the hood! We'll leave the details of this subject for another tutorial.
+
+.. code-block:: python
+
+    # (6) Create a default execution context and a default domain decomposition.
```

Comment:
We refer to it here as an execution context and before as a hardware context, I would stick to execution context in both cases, because it matches the C++ API.